Assignment6.txt

```diff
@@ -0,0 +1,25 @@
+Ashank Kumar
+
+1. num has no value when initialized. So on the second line when the length of the array arr is set to num the array has no value. num is only given an value the 2 lines after where the user is prompted to enter the size of arr, and put a value in arr. Therefore the line: int arr[num] is not in scope and the code will not compile. To fix the code, the line: int arr[num] should put after the scanf. 
+
+1. A segmentation fault is when a program attempts to access memory which it is not allowed. 
```

Comment:
right! as far as your example, that's not considered a segmentation fault. It would just be undefined behavior, not so much a segmentation fault. With that said, an example of a seg fault is the code error from number 1!